Fix Participant tests to use full form flow

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
Fix Participant tests to use full form flow

Before
----------------------------------------
Old way of testing forms

After
----------------------------------------
Full flow used

Technical Details
----------------------------------------
This includes a couple of preliminary PRs that make it look like a lot

Comments
----------------------------------------

eileenmcnaughton committed:
Improve fromEmail address handling

We don't need to validate the from Email address in the postProcess as the
form submit validation expects it to be a supplied value.

Fixing this makes testing easier

eileenmcnaughton committed:
Improve test consistency on capitalisation

eileenmcnaughton committed:
Fix Participant tests to use full form flow

This doesn't get all of them but fixes a bunch

seamuslee001 commented:
@eileenmcnaughton a few test failures here e.g.

```
CRM_Event_Form_ParticipantTest::testSubmit
strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated

/home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/CRM/Event/Form/Participant.php:304
/home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/tests/phpunit/CRM/Event/Form/ParticipantTest.php:350
/home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/tests/phpunit/CRM/Event/Form/ParticipantTest.php:30
/home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviUnitTestCase.php:229
/home/homer/buildkit/extern/phpunit8/phpunit8.phar:1721
```

eileenmcnaughton committed:
Add getParticipantID to avoid non-php8-friendly path check

This particular fail is due to the path being empy in the test environment - But
it's wrong to check it as it is a crufty check for whether
the search task is being used
